active-directory-java-native-headless-v2/README.md

```diff
@@ -0,0 +1,190 @@
+
+- This requires having credentials in the application, which does not happen with the other flows.
+- The credentials should only be used when there is a high degree of trust between the resource owner and the client and when other authorization grant types are not available (such as an authorization code).
+- Do note that this attempts to authenticate and obtain tokens for users using this flow will often fail with applications registered with Azure AD. Some of the situations and scenarios that will cause the failure are listed below  
+  - When the user needs to consent to permissions that this application is requesting.
+  - When a conditional access policy enforcing multi-factor authentication is in force.
+  - Azure AD Identity Protection can block authentication attempts if this user account is compromised.
+  - The user's password is expired and requires a reset.
+
+While this flow seems simpler than the others, applications using these flows often encounter more problems as compared to other flows.
+
+The modern authentication protocols (SAML, WS-Fed, OAuth and OpenID), in principal, discourage apps from handling user credentials themselves. The aim is to decouple the authentication method from an app. Azure AD controls the login experience to avoid exposing secrets (like passwords) to a website or an app.
+
+This enables IdPs like Azure AD to provide seamless single sign-on experiences, enable users to authenticate using factors other than passwords (phone, face, biometrics) and Azure AD can block or elevate authentication attempts if it discerns that the user’s account is compromised or the user is trying to access an app from an untrusted location and such.
+
+## How to run this sample
+
+To run this sample, you'll need:
+
+- Working installation of Java and Maven.
+- An Internet connection.
+- An Azure Active Directory (Azure AD) tenant. For more information on how to get an Azure AD tenant, see [How to get an Azure AD tenant](https://azure.microsoft.com/en-us/documentation/articles/active-directory-howto-tenant/).
+- A user account in your Azure AD tenant. This sample will not work with a Microsoft account (formerly Windows Live account). Therefore, if you signed in to the [Azure portal](https://portal.azure.com) with a Microsoft account and have never created a user account in your directory before, you need to do that now.
+
+## Quick Start
+
+Getting started with the sample is easy. It is configured to run out of the box with minimal setup.
+
+### Step 1: Download Java (8 and above) for your platform
+
+To successfully use this sample, you need a working installation of [Java](http://www.oracle.com/technetwork/java/javase/downloads/index.html) and [Maven](https://maven.apache.org/).
+
+### Step 2:  Clone or download this repository
+
+From your shell or command line:
+
+```shell
+`git clone https://github.com/Azure-Samples/ms-identity-java-desktop.git `
+```
+
+Go to `active-directory-java-native-headless-v2` folder
```

Comment:
Can we rename this folder to "Call-MsGraph-WithUsernamePassword"

Reply:
Sure. I will rename the folder